glib2/ext/glib2/rbglib_regex.c

```diff
@@ -0,0 +1,513 @@
+
+    G_INITIALIZE(self, regex); 
+    return Qnil;
+}
+
+static VALUE
+rg_match_flags(VALUE self)
+{
+  return UINT2NUM(g_regex_get_match_flags(_SELF(self)));
+}
+
+static VALUE
+rg_compile_flags(VALUE self)
+{
+  return UINT2NUM(g_regex_get_compile_flags(_SELF(self)));
+}
+
+static VALUE
+rg_pattern(VALUE self)
+{
+  return CSTR2RVAL(g_regex_get_pattern(_SELF(self)));
+}
+
+static VALUE
+rg_max_backref(VALUE self)
+{
+  return INT2NUM(g_regex_get_max_backref(_SELF(self)));
+}
+
+static VALUE
+rg_capture_count(VALUE self)
+{
+  return INT2NUM(g_regex_get_capture_count(_SELF(self)));
+}
+
+static VALUE
+rg_has_cr_or_lf(VALUE self)
+{
+  gboolean contains_cr_or_lf = g_regex_get_has_cr_or_lf(_SELF(self));
+  
+  if(contains_cr_or_lf == TRUE)
+    return Qtrue;
+  else
+    return Qfalse;
+}
+
+static VALUE
+rg_max_lookbehind(VALUE self)
+{
+  return INT2NUM(g_regex_get_max_lookbehind(_SELF(self)));
+}
+
+static VALUE
+rg_string_number(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string;
+  rb_scan_args(argc, argv, "10", &string);
+  
+  return INT2NUM(g_regex_get_string_number(_SELF(self), RVAL2CSTR(string)));
+}
+
+static VALUE
+rg_split(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string, match_options, array_of_strings;
+  rb_scan_args(argc, argv, "20", &string, &match_options);
+  gchar **strings;
+
+  strings = g_regex_split(_SELF(self),
+                          RVAL2CSTR(string),
+                          NUM2UINT(match_options));
+  
+  array_of_strings = rb_ary_new();
+  gchar **ptr;
+  for(ptr = strings; *ptr != NULL; ptr++)
+  {
+    rb_ary_push(array_of_strings, CSTR2RVAL(*ptr));
+  }
+  
+  g_strfreev(strings);
+  return array_of_strings;
+}
+
+static VALUE
+rg_split_full(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string, start_position, match_options, max_tokens, array_of_strings;
+  GError *error = NULL;
+  rb_scan_args(argc, argv, "40", &string, &start_position, &match_options, &max_tokens);
+  gchar **strings;
+
+  strings = g_regex_split_full(_SELF(self),
+                               RVAL2CSTR(string),
+                               -1,
+                               NUM2INT(start_position),
+                               NUM2UINT(match_options),
+                               NUM2INT(max_tokens),
+                               &error);
+  if(error)
+    RAISE_GERROR(error);
+
+  array_of_strings = rb_ary_new();
+  gchar **ptr;
+  for(ptr = strings; *ptr != NULL; ptr++)
+  {
+    rb_ary_push(array_of_strings, CSTR2RVAL(*ptr));
+  }
+  
+  g_strfreev(strings);
+  return array_of_strings;
+}
+
+static VALUE
+rg_replace(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string, start_position, replacement, match_options;
+  GError *error = NULL;
+  rb_scan_args(argc, argv, "40", &string, &start_position, &replacement, &match_options);
+  gchar *modified_string;
+
+  modified_string = g_regex_replace(_SELF(self),
+                                    RVAL2CSTR(string),
+                                    -1,
```

Comment:
Use `RVAL2CSTR_RAW` and `RVAL2CSTR_LEN` because Ruby's string may not be terminated by `\0`.